### DISCUSSION TITLE
Upcoming breaking changes.

### DISCUSSION THREAD

**Discussion by jheddings** [jheddings/notional#32] [Announcements]
There are several significant breaking changes coming to Notional...

First, we will be moving away from the `from_` factory methods and using an indexer on the metaclass to compose data objects.  Feel free to review the pull request ( #31 ) to see the changes.  This will be released as `v0.3.0`.

Next, we will change the declaration of custom types (`ConnectedPage`'s) to use the `schema` classes rather than the `types`.  This is more consistent with declaring schemas for the non-ORM endpoints and will allow us to perform more natural operations using schema definitions.  Properties will still be modified using the appropriate `types` object.  Please provide comments in the pull request (#33).

Finally, the `connected_page()` method will be improved to accept an existing database object.  This will allow us to generate custom types on the fly from existing databases, rather than having to explicitly map out all the properties in a Notion table.

With these changes in place, Notional will be released as `v0.9.0` and hopefully take us up to the "1.0.0" release.

**Comment by jheddings** [comment]
Also added PR #38 for these changes.

After looking at it a bit, there are a couple of things I'd like to address before getting to `v0.9.0`, mostly around the way connected objects are handled.

**Reply by RickVincent** [reply]
Exciting on such roadmap! Can't wait for 1.0.0!
